packages/trace-viewer/src/sw/snapshotRenderer.ts

```diff
@@ -299,6 +299,25 @@ function snapshotScript(...targetIds: (string | undefined)[]) {
         }
       }
 
+      const canvases = root.querySelectorAll('canvas');
+      if (canvases.length > 0 && screenshotURL) {
+        const img = new Image();
+        img.onload = () => {
+          for (const canvas of canvases) {
+            const context = canvas.getContext('2d')!;
+
+            const boundingRect = canvas.getBoundingClientRect();
+            const xStart = boundingRect.left / window.innerWidth;
+            const yStart = boundingRect.top / window.innerHeight;
+            const xEnd = boundingRect.right / window.innerWidth;
+            const yEnd = boundingRect.bottom / window.innerHeight;
+
+            context.drawImage(img, xStart * img.width, yStart * img.height, (xEnd - xStart) * img.width, (yEnd - yStart) * img.height, 0, 0, canvas.width, canvas.height);
```

Comment:
- Is this going to work nicely for the out-of-viewport canvas? Perhaps we should say "not captured" there instead?
- This code must run after `scrollLeft/scrollTop` are restored in `onLoad()`.
- We should still have a warning somewhere that would explain that image might be wrong.

Reply:
implemented all of it